whois-nrtm4/src/main/java/net/ripe/db/nrtm4/NrtmFileProcessor.java

```diff
@@ -21,38 +21,38 @@ public class NrtmFileProcessor {
 
     private final NrtmFileService nrtmFileService;
     private final NrtmProcessControl nrtmProcessControl;
-    private final NrtmSourceHolder nrtmSourceHolder;
     private final SnapshotFileGenerator snapshotFileGenerator;
+    private final SourceRepository sourceRepository;
 
     public NrtmFileProcessor(
         final NrtmFileService nrtmFileService,
         final NrtmProcessControl nrtmProcessControl,
-        final NrtmSourceHolder nrtmSourceHolder,
-        final SnapshotFileGenerator snapshotFileGenerator
+        final SnapshotFileGenerator snapshotFileGenerator,
+        final SourceRepository sourceRepository
     ) {
         this.nrtmFileService = nrtmFileService;
         this.nrtmProcessControl = nrtmProcessControl;
-        this.nrtmSourceHolder = nrtmSourceHolder;
         this.snapshotFileGenerator = snapshotFileGenerator;
+        this.sourceRepository = sourceRepository;
     }
 
     public void updateNrtmFilesAndPublishNotification() throws IOException {
-        LOGGER.info("runWrite() called");
-        final NrtmSource source = nrtmSourceHolder.getSource();
-        final Optional<SnapshotFile> lastSnapshot = snapshotFileGenerator.getLastSnapshot(source);
-        Optional<PublishableSnapshotFile> publishableSnapshotFile = Optional.empty();
+        LOGGER.info("updateNrtmFilesAndPublishNotification() called");
+        final List<NrtmSourceModel> sourceList = sourceRepository.getAllSources();
+        if (sourceList.isEmpty()) {
+            sourceRepository.createSources();
+        }
```

Comment:
Do we really want to do this? If source is empty it means no initial snapshot exists and it should start creating in jmx enabled. So why create separately?

Reply:
done